Stripe/PublicHeaders/STPSourceParams.h

```diff
@@ -0,0 +1,200 @@
+//
+//  STPSourceParams.h
+//  Stripe
+//
+//  Created by Ben Guo on 1/23/17.
+//  Copyright © 2017 Stripe, Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "STPFormEncodable.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class STPCardParams;
+
+/**
+ *  An object representing parameters used to create a Source object. 
+ *  @see https://stripe.com/docs/api#create_source
+ */
+@interface STPSourceParams : NSObject<STPFormEncodable, NSCopying>
+
+/**
+ *  The type of the source to create. Required.
+ */
+@property (nonatomic, copy, nullable) NSString *type;
```

Comment:
So changing this limits the usage of this api to sources that are known about and manually built in to the SDK. Keeping the raw params as strings gives less type safety, but also allows people to use new sources as they are introduced to the API without us having to release a new SDK version. Do you think that is a fair tradeoff? I have some mixed feelings and could go either way.